public/app/features/templating/formatRegistry.ts

```diff
@@ -244,7 +245,13 @@ export const formatRegistry = new Registry<FormatRegistryItem>(() => {
       description:
         'Format variables as URL parameters. Example in multi-variable scenario A + B + C => var-foo=A&var-foo=B&var-foo=C.',
       formatter: (options, variable) => {
+        const { isScoped, value } = options;
         const { name, type } = variable;
+
+        if (isScoped) {
+          return formatQueryParameter(name, value);
+        }
+
```

Comment:
This is confusing, why would we care about if the value is scoped or not here? I do not understand this formatter, the formatter options parameter has a value property, why is this formatter not using that?

Reply:
I understand its confusing, this is a special formatter because when it's not used outside a scope setting then you want it to resolve all the values (for a multivalued variable), i.e. var1=value1&var1=value2 but in a scoped setting you want it to resolve to var1=value1.

Reply:
An alternative would be to make sure to set the "correct" value for non scoped vars for this specific formatter here https://github.com/grafana/grafana/blob/8722c19cc396e49ab2da8142594d9bdbb8bd10af/public/app/features/templating/template_srv.ts#L297. Maybe that is a better approach?

Reply:
> Itss is a special formatter because when it's not used outside a scope setting then you want it to resolve all the values 

But that is true for all formatters, so not sure how that makes this special. The only thing special with this formatter is that it needs to know the name of the variable not just the value

The multi value is passed via options.value, so as long as this formatter always takes the value from there it should work for both scoped and non scoped uses

Reply:
Thanks for the feedback, I've updated the code.